### DISCUSSION TITLE
Will a workflow be created after i push a yml file under .github/workflows no matter which branch i was?

### DISCUSSION THREAD

**Discussion by yangjuexi** [community/community#28461] [Actions] [answered]
I was confused about when a workflow will be created.

1. In the web, if i click the button 'new workflow' and then commit the yml file to master or a new branch, a workflow will be created.
2. If i create a yml file on a new branch and push it form local repository, a workflow also be created
3. but, if i create a yml file on a exist branch and push it to GitHub , none workflow will be created
4. i have read the docs but found nothing about how to create a workflow.just a example workflow
5. also, i have google this question but have no idea still.
So, my question is :when a workflow will be created ?

**Comment by airtower-luna** [comment, marked as answer]
It depends on the kind of event trigger(s) you define in the `on` section of your workflow file (the YAML file). Some of them work on any branch, for example `push` (as long as you don't also set a branch filter that excludes the current branch). For other events the workflow file must be on the default branch to work.

See the full [list of events](https://docs.github.com/en/actions/using-workflows/events-that-trigger-workflows) for how it is in each case, specifically look at the listed `GITHUB_REF`.

**Reply by yangjuexi** [reply]
Sorry to reply after a long time and thanks for your reply.
I just want to create a manually triggered workflow on an exist feature branch. It didn't work after i pushed the .yaml file.That's why i ask this question.Finally I merge the branch to master(that's what you said default branch) and it works although i still don't know the accurate opportunity. Your answer makes me clear. Thanks again.
